dadi/frontend/components/DocumentFilter/DocumentFilter.jsx

```diff
@@ -20,29 +25,63 @@ export default class DocumentFilter extends Component {
       '$lt': 'Less than',
       '$lte': 'Less than or Equal to'
     }
+    this.state = {
+      field: null,
+      type: '$eq',
+      value: null
+    }
+  }
+
+  componentWillMount() {
+    const {field, value, fields} = this.props
+
+    // Evaluate passed filter / store in state
+    if (field && value) {
+      this.setState({
+        field: field,
+        type: (typeof value === 'string') ? '$eq' : Object.keys(value)[0],
+        value: this.filterValue(value)
+      })
+    }
   }
 
   render() {
-    const { field, value, fields } = this.props
+    const {fields, value} = this.props
+
     return (
       <div>
-        <select>
+        <select id="field" onChange={this.onChange.bind(this)}>
+          <option disabled selected value>Select field</option>
           {Object.keys(fields).map(key => (
-            <option selected={field === key} name={key}>{fields[key].label}</option>
+            <option selected={this.state.field === key} key={key} value={key}>{fields[key].label}</option>
           ))}
         </select>
-        <select>
+        <select id="type" onChange={this.onChange.bind(this)}>
+          <option disabled selected value>Select a type</option>
           {Object.keys(this.filterTypes).map(key => (
-            <option name={key}>{this.filterTypes[key]}</option>
+            <option selected={this.state.type === key} key={key} value={key}>{this.filterTypes[key]}</option>
           ))}
         </select>
-        {field && value && (
-          <TextInput value={this.filterValue(value)} />
+        {this.state.field && this.state.type && (
+          <TextInput id="value" value={this.state.value} onChange={this.onChange.bind(this)} />
```

Comment:
Is it so that you can use `onChange` with different elements? If so, I'd suggest using the second argument of `.bind()`, as used [here](https://github.com/dadi/publish/blob/d0c5b545785a9115ed3634c55e34befce8d01930/dadi/frontend/components/Header/Header.jsx#L78), which will then come through as the first argument of `onChange`.